src/scrapers/dns.iced

```diff
@@ -97,13 +97,23 @@ exports.DnsScraper = class DnsScraper extends BaseScraper
 
   # ---------------------------------------------------------------------------
 
+  _resolve_txt : (domain, cb) ->
+    # We can use a DNS library passed in (in the case of native-dns running on
+    # the server)
+    dnslib = @libs.dns or dns
+    try
+      dnslib.resolveTxt domain, (err, records) ->
```

Comment:
i didn't realize you could try/catch callback functions like this, cool!

Reply:
you should also be able to do something like
```
try
   await foo defer err, res
catch err
cb err, res
```
but is silently broken in iced2 😢 IIRC the code after catch is not emitted at all for some reason

Reply:
This only catches exceptions thrown in the sync part of `resolveTxt` though, if theres some exception in some callback afterwards, or even in the callback we supply here, it's not going to be caught by the try block.